pandas/core/indexes/datetimelike.py

```diff
@@ -854,6 +854,116 @@ def _concat_same_dtype(self, to_concat, name):
             new_data = np.concatenate([c.asi8 for c in to_concat])
         return self._simple_new(new_data, **attribs)
 
+    def _slice_by_value(self, start, end):
```

Comment:
instead of doing this, simply return the sliced; move the ``_shallow_copy`` to intersection, then you don't need to pass params around